packages/reports/addon/components/filter-builders/date-time.js

```diff
@@ -1,51 +1,178 @@
    */
-  supportedOperators: computed(function() {
+  @computed('dateTimePeriodName')
+  get supportedOperators() {
     return [
+      {
+        id: 'current',
+        longName: `Current ${this.dateTimePeriodName}`,
+        valuesComponent: 'filter-values/current-period'
+      },
+      {
+        id: 'inPast',
+        longName: 'In The Past',
+        valuesComponent: 'filter-values/lookback-input'
+      },
+      {
+        id: 'since',
+        longName: 'Since',
+        valuesComponent: 'filter-values/since-input'
+      },
       {
         id: 'in',
-        longName: 'In Range',
+        longName: 'Between',
         valuesComponent: 'filter-values/date-range'
+      },
+      {
+        id: 'advanced',
+        longName: 'Advanced',
+        valuesComponent: 'filter-values/advanced-interval-input' // TODO: Show two inputs and range
       }
     ];
-  }),
+  }
 
   /**
    * @property {Object} requestFragment - interval fragment from request
    */
-  requestFragment: undefined,
+  requestFragment = undefined;
+
+  /**
+   * Finds the appropriate interval operator to modify an existing interval
+   * @param {Interval} interval - the interval to choose an operator for
+   */
+  operatorForInterval(interval) {
+    const { start, end } = interval.asStrings();
+
+    const [, /*match*/ lookback /*amount*/, , dateTimePeriodLabel] = /(P)(\d+)(\w)/.exec(start) || [];
+    let operatorId;
+    if (start === 'current' && end === 'next') {
+      operatorId = 'current';
+    } else if (lookback === 'P' && ['D', 'W', 'M', 'Y'].includes(dateTimePeriodLabel) && end === 'current') {
+      operatorId = 'inPast';
+    } else if (moment.isMoment(interval._start) && (end === 'current' || end === 'next')) {
+      operatorId = 'since';
+    } else if (moment.isMoment(interval._start) && moment.isMoment(interval._end)) {
+      operatorId = 'in';
+    } else {
+      operatorId = 'advanced';
+    }
+
+    return this.supportedOperators.find(op => op.id === operatorId);
+  }
+
+  /**
+   *
+   * @param {Interval} interval
+   * @param {String} newOperator
+   */
+  convertIntervalToOperator(interval, dateTimePeriod, newOperator) {
+    const { start, end } = interval.asMomentsForTimePeriod(dateTimePeriod);
+
+    if (newOperator === 'current') {
+      return Interval.parseFromStrings('current', 'next');
+    } else if (newOperator === 'inPast') {
+      const isQuarter = dateTimePeriod === 'quarter';
+      const intervalDateTimePeriod = isQuarter ? 'month' : dateTimePeriod;
+
+      let intervalValue;
+      if (end.isSame(moment(getFirstDayOfIsoDateTimePeriod(moment(), dateTimePeriod)))) {
+        // end is 'current', get lookback amount
+        intervalValue = interval.diffForTimePeriod(intervalDateTimePeriod);
+      } else {
+        intervalValue = 1;
+      }
+
+      if (isQuarter) {
+        // round to quarter
+        const quarters = Math.max(Math.floor(intervalValue / 3), 1);
```

Comment:
Can we do
`const MONTHS_IN_A_QUARTER = 3;` 
in whichever files this sort of conversion is done? It just makes it a little easier to read.